### DISCUSSION TITLE
NEC Vxx x87 FPU emulation

### DISCUSSION THREAD

**Discussion by Torinde** [86Box/86Box#3399] [Ideas]
Happy to see support for NEC V20 and its 8080 mode. As shown at [2](https://www.cpushack.com/2021/09/01/necs-forgotten-fpus/),[3](https://www.cpu-world.com/forum/viewtopic.php?t=12863&view=previous),[1](https://www.vogons.org/viewtopic.php?t=68413) - there were also an accompanying FPU(s): NEC μPD72091, μPD72191, D9008D supporting extra x87 instructions: FTAN, FATAN, FTANH, FXP2, FXPT, FXPE, FLGTX, FLGEX, FREM, FLDDTR, FSAVEA, FNSAVEA, FRSTORA, FSTENVA, FNSTENVA, FLDENVA, FLDCWA, FPOWER.

Also interesting are the CPUs with 24-bit addresses:
- NEC V33/V53 - extended addressing mode via `BRKXA`/`RETXA` instructions
- NEV V55 - [extended segment registers](https://www.renesas.com/en/document/dst/upd70433-data-sheet#page=23)

**Comment by OBattler** [comment]
The 8080 mode is also not emulated.

**Comment by Torinde** [comment]
Z80 emulation is part of the IBM Music Feature card patch for DOSbox: joncampbell123/dosbox-x/issues/3596

**Comment by OBattler** [comment]
Converting this to ideas since it's not sure the produced chips work with the V20/V30.

**Reply by Torinde** [reply]
"As shown at [2](https://www.cpushack.com/2021/09/01/necs-forgotten-fpus/)" - per the link from my post above it seems they work with V20/V30:

"The upd72191 was made in CMOS and is a bit like an enhanced 80C187 but with **support for the V20/V30**."

**Comment by starcrescent** [comment]
Why there is no 8 MHz frequency for NEC V20 CPU?
There is 4.77, 7.16, 10, 12 and 16 but no 8 MHz.
